### ISSUE TITLE
typing in the notification box does not work in Wayland

### ISSUE THREAD

**Issue by greatquux** [muffinmad/pidgin-im-gnome-shell-extension#32]
I figured I'd do a little testing when running GNOME Shell on Wayland (launched from gdm3 not lightdm). The notification pops up, and seems to act normal (you can even re-open it and it stays up to date with new messages) but you can't type into it at all.  It seems it never gains the focus as my keystrokes go to the window underneath. I can click in it to get a cursor but can't type.   

This is probably not your problem, and it will be a while before you have to worry about it anyway, I figured I would let you know.

**Comment by greatquux** [comment]
I should also mention this testing was done under Ubuntu GNOME 16.04, so that's using GNOME 3.18, and this might be fixed under GNOME 3.20 as I think they have fixed some other extensions bugs there.  Right now the gnome-staging PPA that has 3.20 for 16.04 is a little unstable, once it stabilizes I'll try to check it out there (or just wait for 16.10).

**Comment by muffinmad** [comment]
Thanks. And 3.20 works the same way, unfortunately. At least on Arch. If empathy notifications works the same way then it's must be gnome issue.

**Comment by Thisuu** [comment]
Hi, I can confirm that the same happens on debian under Wayland/Gnome 3.20. Plugin works fine but i can't type anything.

**Comment by greatquux** [comment]
I guess the question really becomes is it a problem with muffinmad's extension?  Or a general problem with GNOME extensions on wayland not being able to accept keyboard input, maybe due to additional security restrictions on Wayland, and is there any way around that?

**Comment by CyberPsyko** [comment]
Same here (Fedora 24/Gnome 3.20 & Wayland). I've tried to find similar issues with Gnome notification typing areas, but I haven't been able to find anything. This is very frustrating.

**Comment by muffinmad** [comment]
Try it on Gnome 3.22, please.

**Comment by greatquux** [comment]
Just upgraded to Ubuntu 17.04 development release today (I got sick of working around an evolution issue that had been fixed but not published to 16.10 as is the usual way of things).  Delighted to report that this WORKS in GNOME 3.22 Wayland session here!  I'll do further testing later but just sending a message to myself and then popping up the notification I found I could type.
![itworks](https://cloud.githubusercontent.com/assets/1655942/21067654/ba0c5bca-be3a-11e6-9f1b-53307952ea9e.png)
